### ISSUE TITLE
opam 2.0.0~rc2 prints a message on every prompt

### ISSUE THREAD

**Issue by diml** [ocaml/opam#3363]
I'm just upgraded to opam 2.0.0~rc2 and I now get a message before every prompt as follow:

```
[NOTE] To make opam select /usr/local/home/jdimino/opam-root as its root in the current shell, add --set-root or set OPAMROOT
$ true
[NOTE] To make opam select /usr/local/home/jdimino/opam-root as its root in the current shell, add --set-root or set OPAMROOT
$ true
[NOTE] To make opam select /usr/local/home/jdimino/opam-root as its root in the current shell, add --set-root or set OPAMROOT
$ true
```

I do set `OPAMROOT` in my `~/.bashrc` so I'm not sure what the issue is. This is what I have in my `~/.bashrc`:

```bash
export OPAMROOT="$HOME/code/opam-root"
source "$OPAMROOT/opam-init/init.sh"
```

`opam config env` prints the same message.

**Comment by rjbou** [comment]
You have the shell hook enabled. Did you configure it by choosing `bashrc` or the default one?

**Comment by diml** [comment]
I didn't configure anything while upgrading. I'm trying with a fresh root to see if I get the same error, however it seems odd to me that `opam config env` prints this message as well.

**Comment by diml** [comment]
I get the same error with a fresh root. I did the following:

- I removed the opam configuration from my ~/.bashrc file
- I opened a new shell
- I ran `opam init --root ~/code/toto` and answered yes to both questions
- I opened a new shell

In the new shell, I get the following message before every prompt:

```
[ERROR] Opam has not been initialised, please run `opam init'
```

Then I edited my `~/.bash_profile` to add `export OPAMROOT=$HOME/code/toto` before the code inserted by opam and opened a new shell and I now get the following message before every prompt:

```
[NOTE] To make opam select /usr/local/home/jdimino/toto as its root in the current shell, add --set-root or set OPAMROOT
```

**Comment by AltGr** [comment]
ugh, quite weird, sorry!
You can at least disable the auto-eval feature to avoid the nag at every prompt with `opam init --disable-shell-hook` (and restarting the shell)

**Comment by AltGr** [comment]
Ah, looking at the code, there may be one normalisation missing (`OPAMROOT` is compared non-normalised with the current root, which is normalised). I'll fix it, but to confirm, did you have any symbolic links in there ?

You could check with `opam env --set-root --root xxx | grep OPAMROOT`

**Comment by diml** [comment]
There is indeed a symbolic link, `~/code` is a symlink to `/usr/local/home/jdimino` and `opam env --set-root --root ~/code/opam-root` does returned the normalized path:

```
$ export OPAMROOT=~/code/opam-root; opam env --set-root --root ~/code/opam-root|grep OPAMROOT
OPAMROOT='/usr/local/home/jdimino/opam-root'; export OPAMROOT;
```

In the meantime I tried to replace setting `OPAMROOT` by making `~/.opam` a symlink and the message seems to be gone.

BTW, I'm doing this as my home directory is on nfs.

**Comment by diml** [comment]
No problem, thanks for the quick fix! I'm going to try it now.

**Comment by AltGr** [comment]
Fixed by #3364
